### DISCUSSION TITLE
How do I set up development environment

### DISCUSSION THREAD

**Discussion by ademDurakovic** [uNetworking/uWebSockets#1726] [Q&A] [answered]
Hi, apologies for dumb question but how do I even get this set up so I can start using it?

Currently I clone the repo and have asked GPT so give me some next instructions but those are not working and I don't really see anything in the ReadMe or UserManual a step by step on how to get it integrated into my IDE. I am on MAC if that helps.

I appreciate the help.

**Comment by uNetworkingAB** [comment]
> how to get it integrated into my IDE

How would I know? ;)

**Reply by ademDurakovic** [reply]
Hi sorry haha let me give you some more information maybe?

I git cloned recursively and then I ran make in USockets then UwebSockets directories.

I am using Clion for my IDE and so I would assume I just like these folders into my cmake list right?

currently mine looks like :
cmake_minimum_required(VERSION 3.24)
project(uWebSocket)

set(CMAKE_CXX_STANDARD 17)

# Add the directory containing uSockets headers
include_directories(/Users/adem/uWebSockets/uSockets/src)

# Add your source files
set(SOURCE_FILES
        main.cpp
        )

# Add the executable
add_executable(${PROJECT_NAME} ${SOURCE_FILES})

# Link against the uSockets library
target_link_libraries(${PROJECT_NAME} /Users/adem/uWebSockets/uSockets/uSockets.a)

But when I try to include the header file I see in examples it just tells me it cannot find it.

I appreciate the help :)

**Reply by ademDurakovic** [reply]
Also, I plan on using this library for a school project where I am being forced to use 2 languages and 1 needs to be C++. We chose react for the front end, I plan on using this library to connect the 2 and have them communicate. I am in the right place for that?

**Reply by uNetworkingAB** [reply, marked as answer]
Something like **include_directories(/Users/adem/uWebSockets/src)** would also be needed

**Reply by ademDurakovic** [reply]
I really appreciate the help. I'm sure I'll have more questions later lol but currently I have some pretty simple code running and it seems to work now. 

thank you
